### PR TITLE
Heroku Questions

### PR DESCRIPTION
**Pull request by EdenVicary**
- Dynos
- Scaling
- Messaging

### CODE REVIEW

File: questions.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
-
+1. So am I correct in thinking that [scaling dynos](https://devcenter.heroku.com/articles/how-heroku-works#running-applications-on-dynos) is horizontally scaling?
```
Comment (felixclack) [inline questions.md]: Yes

File: questions.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
...
+  - To scale horizontally (scale out), add more dynos. For example, adding more web dynos allows you to handle more concurrent HTTP requests, and therefore higher volumes of traffic. For more information, see Scaling Your Dyno Formation.
```
Comment (felixclack) [inline questions.md]: Exactly

File: questions.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
...
+To scale vertically (scale up), use bigger dynos. The maximum amount of RAM available to your application depends on the dyno size you use. For more information, see Dyno Size.
```
Comment (felixclack) [inline questions.md]: Yep

File: questions.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
...
+2. What are some really good ways to use [worker dynos](https://devcenter.heroku.com/articles/dynos#types-of-dynos)?
```
Comment (felixclack) [inline questions.md]: The most common ones are the `worker` process. Which could be a background worker that processes the job queue.

Also, `clockwork` is a cron equivalent on Heroku. Trigger code at specific times of the day.

File: questions.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
...
+3. [One-off Dynos](https://devcenter.heroku.com/articles/one-off-dynos)?
```
Comment (felixclack) [inline questions.md]: Running `heroku run console` fires up a dyno with `rails console` running in it. It shuts down when you exit.

File: questions.md [hunk 6]
```diff
@@ -1,3 +1,8 @@
...
+4. Is there a Rails/Heroku way on how to do queing/is it a thing?
```
Comment (felixclack) [inline questions.md]: Rails has a built in queue abstraction http://guides.rubyonrails.org/active_job_basics.html.

You can back it with something like https://github.com/resque/resque or https://github.com/mperham/sidekiq

File: questions.md (hunk 4 above)
Comment (EdenVicary) [inline questions.md]: - So clock work calls the cron on a worker?
- Are workers the same code base ... just called in a different way?

File: questions.md (hunk 5 above)
Comment (EdenVicary) [inline questions.md]: I just had a go that's pretty neat.

File: questions.md (hunk 6 above)
Comment (EdenVicary) [inline questions.md]: Soooooo, good

**Comment by EdenVicary** [comment]
So this Pull request uses the originally suggested method of submitting questions to a .md file, this has since been deprecated. We will pose questions as github issues.

File: questions.md (hunk 4 above)
Comment (felixclack) [inline questions.md]: Yeah, same code, but running a different process. Perhaps a rake task or maybe a file in `bin/`.

Comment (EdenVicary) [inline questions.md]: Awesome

**Pull request closed without merging**